lib/utils/fs_test.go

```diff
@@ -38,7 +40,32 @@ func TestOpenFileLinks(t *testing.T) {
 	// broken-s -> nonexistent
 	// hardfile
 	// hardfile-h -> hardfile
-	rootDir := t.TempDir()
+
+	// macOS is special cased here since t.TempDir() returns a path to a protected directory that doesn't allow symlinks.
+	var rootDir string
+	switch runtime.GOOS {
+	case "darwin":
+		suffix := make([]byte, 8)
+		_, err := rand.Read(suffix)
+		if err != nil {
+			t.Fatalf("failed to generate random suffix: %v", err)
+		}
+
+		rootDir = "/private/tmp/" + "teleport-test-" + hex.EncodeToString(suffix)
+		err = os.Mkdir(rootDir, 0700)
```

Comment:
I think you can replace all of this with

```
os.MkdirTemp("/private/tmp/teleport-test-*")
```

Reply:
thanks, didn't know that existed

Reply:
Fixed

Reply:
You don't need to generate a random suffix anymore though. That's what MkdirTemp will do for you.

Reply:
Just forgot to push

Reply:
Done